STEP WISE RAMPING ADD-ON

jgeab opened the pull request:
* Introduction of the STEPWISE RAMPING add-on

Other changes:
* Restructuring of the code to allow for different scaling of the timeseries while respecting annual constraints (Future task: to allow to define seasonal constraints instead (something like BB3))
* Small correction related to investments allowance in unit commitment and ramping_step_wise add-on
* OUTPUT_SUMMARY.inc simplified and improved
* Adding the possibility to run a different Balmorelbb4.sim
* Error correction in QGEKNT of the old code, found by Christoph Kumbf
* Including explanation of the loop inside stepwise_ramping_ipardecdef.inc
* Excluding Output_summary and ramping add-on as default
* Renaming files and folders
* Small correction of IHOURSINT redefinition

jgeab committed:
STEP WISE RAMPING ADD-ON
* Restructuring of the code to allow for different scaling of the timeseries while respecting annual constraints (Future task: to allow to define seasonal constraints instead (something like BB3))
* Small correction related to investments allowance in unit commitment and ramping_step_wise add-on
* OUTPUT_SUMMARY.inc simplified and improved
* Adding the possibility to run a different Balmorelbb4.sim
* Error correction in QGEKNT of the old code, found by Christoph Kumbf
* Including explanation of the loop inside stepwise_ramping_ipardecdef.inc
* Excluding Output_summary and ramping add-on as default
* Renaming files and folders
* Small correction of IHOURSINT redefinition

fraukewiese reviewed (commented):
- Great to have this option
- Please in the future, do not mix the addition of an add-on with bug-fixes and output_summary and "small corrections. Bug fixes, corrections should be a different pull request than a new addon.
- what means "renaming files and folders?
- why is there a line deleted in Balmorelbb4.sim about hydro inflow to reservoirs?

jgeab commented:
Dear Frauke, 
*I will try harder in the future not do it. For some reason I feel extremely tempted to do so. I apologize :(
*I have renamed the ramping addon, as well as the ramping part of the UC to have less confusing names
*The line deleted from BalmorelBB4.sim is moved to Balmorelbb4.inc. The purpose of doing this was to gather the scaling of all timeseries together, so the option regarding which scaling to be used in the balopt.opt file would work fine. I changed and some other stuff regarding the definition of IHOURSINT to scale the timeseries. Sorry for mixing it, since this has nothing to do with the ramping add-on :(

fraukewiese commented:
- I am fine for the ramping Addon and would approve that.
- I am however a bit worried about "some other stuff regarding the definition of IHOURSINT to scale the timeseries" and moving stuff from "BalmorelBB4.sim to Balmorelbb4.inc" since that seems to be at the heart of the model and I do not fully oversee the consequences. Could @IdaGJ and @HansRavn have a look at that and comment on that?
- If you want to have a fast approval for the addon, better split the pull request.
